KerbalStuff/blueprints/profile.py

```diff
@@ -1,53 +1,77 @@
             if not current_user.admin:
                 abort(403)
     forum_url_username = None
     forum_url = None
-    match = FORUM_PROFILE_URL_PATTERN.match(user.forumUsername)
+    match = FORUM_PROFILE_URL_PATTERN.match(profile.forumUsername)
     if match:
         forum_url_username = match.group('name')
-        forum_url = (re.sub('^http://', 'https://', user.forumUsername)
+        forum_url = (re.sub('^http://', 'https://', profile.forumUsername)
                      if match.group('prefix') else
-                     f'https://{user.forumUsername}')
-    show_unpublished = current_user and (current_user.id == user.id or current_user.admin)
+                     f'https://{profile.forumUsername}')
+    show_unpublished = current_user and (current_user.id == profile.id or current_user.admin)
     # Get the mods grouped by game, as [(game1_name, [mod1, mod2, ...]), (game2_name, [...]), ...]
     # with the games sorted alphabetically and the mods sorted by age.
     mods_created = list(map(lambda grp: (grp[0], sorted(grp[1],
                                                         key=lambda m: m.created,
                                                         reverse=True)),
-                            groupby(sorted(user.mods if show_unpublished
+                            groupby(sorted(profile.mods if show_unpublished
                                            else filter(lambda m: m.published,
-                                                       user.mods),
+                                                       profile.mods),
                                            key=lambda m: m.game.name),
                                     lambda m: m.game.name)))
-    mods_followed = sorted(user.following, key=lambda mod: mod.created, reverse=True)
+    mods_followed = sorted(profile.following, key=lambda mod: mod.created, reverse=True)
     return render_template("view_profile.html",
-        profile=user, forum_url=forum_url, forum_url_username=forum_url_username,
-        mods_created=mods_created, mods_followed=mods_followed)
+                           profile=profile, forum_url=forum_url, forum_url_username=forum_url_username,
+                           background=profile.background_url(_cfg('protocol'), _cfg('cdn-domain')),
+                           mods_created=mods_created, mods_followed=mods_followed)
+
+
+@profiles.route("/profile/<username>/background")
+def profile_background(username: str) -> werkzeug.wrappers.Response:
+    profile = User.query.filter(User.username == username).first()
+    if not profile:
+        abort(404)
+    if not profile.public:
+        if not current_user:
+            abort(401)
+        if current_user.username != profile.username:
+            if not current_user.admin:
+                abort(403)
+
+    storage = _cfg('storage')
+    if not storage:
+        abort(404)
+
+    return sendfile(os.path.join(storage, profile.thumbnail), False)
```

Comment:
`profile.thumbnail` throws an exception.

![image](https://user-images.githubusercontent.com/1559108/127686186-11fcb33a-763c-48bd-be5c-ec0cac1a32d5.png)

I assume this is left over from an earlier revision where that column would have been added.

Reply:
Oh yeah, that should be `profile.backgroundMedia`. Must have accidentally changed that when refactoring for the mod thumbnails. Fortunately users don't have thumbnails, that would make it all even more complicated.

I also have to take a look at that `os.path.join` again, I'm not sure if it belongs there.